File: vehicle-rentals-microservices/vendor-service/src/main/java/com/project/vendor_service/entity/Vendor.java
package com.project.vendor_service.entity;

import java.time.LocalDateTime;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data

@Entity
public class Vendor {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long vendorId;
	private String name;
	private String email;
	private String password;
	private String phone;
	private LocalDateTime createdAt;
	private LocalDateTime updateAt;
	private LocalDateTime deletedAt;

	@PrePersist
	public void updateDate() {
		this.createdAt=LocalDateTime.now();
		this.updateAt=LocalDateTime.now();
	}
}

File: onthego/src/app/models/vendor.ts
export class Vendor {
    vendorId: number;
    name: string;
    email: string;
    password: string;
    phone: string;
    createdAt: Date;
    updateAt: Date;
    deletedAt?: Date; 

    constructor(
        vendorId: number,
        name: string,
        email: string,
        password: string,
        phone: string,
        createdAt: Date,
        updateAt: Date,
        deletedAt?: Date
    ) {
        this.vendorId = vendorId;
        this.name = name;
        this.email = email;
        this.password = password;
        this.phone = phone;
        this.createdAt = createdAt;
        this.updateAt = updateAt;
        this.deletedAt = deletedAt;
    }
}
